sources/trophy.move

```diff
@@ -0,0 +1,310 @@
+            // For `name` one can use the `TrophyNFT.name` property
+            string::utf8(b"Artfi"),
+            // Description is static for all `TrophyNFT` objects.
+            string::utf8(b"Artfi NFT"),
+        ];
+
+        // Claim the `Publisher` for the package!
+        let publisher = package::claim(otw, ctx);
+
+        // Get a new `Display` object for the `TrophyNFT` type.
+        let display_object = display::new_with_fields<TrophyNFT>(
+            &publisher, keys, values, ctx
+        );
+
+        // Commit first version of `Display` to apply changes.
+        display::update_version(&mut display_object);
+
+        transfer::public_transfer(publisher, tx_context::sender(ctx));
+        transfer::public_transfer(display_object, tx_context::sender(ctx));
+
+        transfer::share_object(NFTInfo {
+            id: object::new(ctx),
+            name: string::utf8(b"Artfi"),
+            user_detials: vec_map::empty<ID, Attributes>(),  
+        });
+
+        transfer::share_object(NftCounter{
+            id: object::new(ctx),
+            count: vec_map::empty<address, u64>()
+        });
+
+        transfer::transfer(AdminCap {
+            id: object::new(ctx)
+        }, tx_context::sender(ctx));
+    }
+
+    // ===== Public view functions =====
+
+    /// Get the NFT's `name`
+    public fun name(nft: &TrophyNFT): &String {
+        &nft.name
+    }
+
+    /// Get the NFT's `url`
+    public fun url(nft: &TrophyNFT): &Url {
+        &nft.url
+    }
+
+    /// Get the NFT's `ID`
+    public fun id(nft: &TrophyNFT): ID {
+        object::id(nft)
+    }
+
+    /// Get Attributes of NFT's
+    public fun attributes(nft: &TrophyNFT, nft_info: &NFTInfo): Attributes{
+        *(vec_map::get(&nft_info.user_detials, &object::id(nft)))
+    }
+
+    /// Get shipment status of NFT's
+    public fun shipment_status(nft: &TrophyNFT, nft_info: &NFTInfo): String {
+        vec_map::get(&nft_info.user_detials, &object::id(nft)).shipment_status
+    }
+
+    // === Public-Mutative Functions ===
+
+    /// Create a new Trophy
+    public entry fun mint_nft(
+        nft_info: &mut NFTInfo,
+        mint_counter: &mut NftCounter,
+        user: address,
+        url: vector<u8>,
+        ctx: &mut TxContext
+    ) { 
+        check_mint_limit(mint_counter, user);
+
+        let id: ID = mint_func(
+            nft_info,
+            url,
+            user,
+            ctx
+        );
+
+        base_nft::emit_mint_nft(id, tx_context::sender(ctx), nft_info.name);
+    }
+
+    /// Permanently delete `NFT`
+    public entry fun burn(nft: TrophyNFT, nft_info: &mut NFTInfo, _: &mut TxContext) {
+        let _id = object::id(&nft);
+        let (_burn_id, _burn_attributes) = vec_map::remove(&mut nft_info.user_detials, &_id);
+        
+        let TrophyNFT { id, name: _, url: _ } = nft;
+        object::delete(id);
+    }
+
+    // === AdminCap Functions ===
+    
+    /// Create a multiple Trophy
+    public fun mint_nft_batch(
```

Comment:
User mints token, we dont need batching here. Lets keep one function only either mint NFT or batch mint NFT, but both should take fraction ID as a parameter and should ensure that NFT has not been minted already for the passed value.